### PR TITLE
Finish hiding TTEntry pointers (as started by 5345/5369)

### PR DESCRIPTION
**Pull request by dubslow**
Passed STC: https://tests.stockfishchess.org/tests/view/667e26bd80450dba9659134c
LLR: 2.94 (-2.94,2.94) <-1.75,0.25>
Total: 178624 W: 46470 L: 46407 D: 85747
Ptnml(0-2): 607, 20409, 47203, 20500, 593

The "simple" version failed two STCs:
https://tests.stockfishchess.org/tests/view/66710959602682471b065043 https://tests.stockfishchess.org/tests/view/66744bd8602682471b065292

no functional change

### CODE REVIEW

**Comment by vondele** [comment]
so prefetch is not constexpr according to clang, and while that's debatable, it means we can't merge the PR as is. If this can't be fixed, please close.

**Comment by dubslow** [comment]
At this point I'm bumping right up on the edge of my knowledge/experience, so I'd appreciate a bit of help/advice.

I guess by your message that this was at least a reasonable try.

Would it be worthwhile to "degrade" prefetch from `constexpr` to merely `inline`, and fishtest that instead? Or would that be a waste in some way?

**Comment by vondele** [comment]
I guess just declaring it inline (or even omitting that) is just fine. No need to fishtest that particular patch. However, it looks like you have now 3 TT related PR pending. I would suggest you create the combined one for that and test that so you show where you'd like to get to with these rewrites.

**Comment by Sopel97** [comment]
`constexpr` on prefetch makes no sense anyway

**Comment by dubslow** [comment]
1) Given that the simplest, obvious version failed two STCs, I'm quite leery about making any change to this PR untested. Those failed tests indicate, to me at least, that even teeny, seemingly-trivial tweaks might compile noticeably differently. If you say that changing `constexpr prefecth()` to `inline prefetch()` is fine without test, then I'll do that.

2) As far as I can figure my three PRs are unrelated.

* One is strictly about commentary of `tt.*`, not code.

* One is about design/encapsulation previously suggested by Sopel relating to `search.cpp` (my further test in that direction had partially failed, partially passed, I was gonna ask for opinions about that),

* and this TTEntry pointer thing is not related to either commentary or Sopel's search encapsulation, but rather just completing the theme that locutus started (and you agreed was desirable at the time).

**Comment by dubslow** [comment]
Per the msg, I've forcepushed to respell `constexpr void prefetch()` as `inline void prefetch()`

File: src/tt.cpp [hunk 1]
```diff
@@ -241,9 +241,10 @@ std::tuple<bool, TTData, TTWriter> TranspositionTable::probe(const Key key) cons
     return {false, replace->read(), TTWriter(replace)};
 }
 
-
+// This points to the whole array contained within the `Cluster`
```
Comment (MinetaS) [inline src/tt.cpp:244]: This comment is not necessary.

File: src/tt.cpp [hunk 2]
```diff
@@ -241,9 +241,10 @@ std::tuple<bool, TTData, TTWriter> TranspositionTable::probe(const Key key) cons
...
 TTEntry* TranspositionTable::first_entry(const Key key) const {
     return &table[mul_hi64(key, clusterCount)].entry[0];
 }
 
+
```
Comment (MinetaS) [inline src/tt.cpp:249]: Unnecessary line here...

File: src/tt.h [hunk 3]
```diff
@@ -81,15 +82,19 @@ class TranspositionTable {
     uint8_t generation() const;  // The current age, used when writing new data to the TT
     std::tuple<bool, TTData, TTWriter>
     probe(const Key key) const;  // The main method, whose retvals separate local vs global objects
-    TTEntry* first_entry(const Key key)
-      const;  // This is the hash function; its only external use is memory prefetching.
+    // Prefetch the cacheline which includes this key's entry
```
Comment (MinetaS) [inline src/tt.h:85]: Also not really necessary comment.

**Comment by vondele** [comment]
I will close the related 
#5419 
#5395 
#5389 

I suggest to put all TT related changes in a single PR after retesting for easier review.

**Pull request closed without merging**